### PR TITLE
ARROW-14013: [C++][Docs] Add instructions for Fedora

### PR DESCRIPTION
**Pull request by bkmgit**
Dependencies for building on Fedora Linux

### CODE REVIEW

**Comment by kou** [comment]
Could you open a JIRA issue and associated with this pull request?

See also the comment from bot: https://github.com/apache/arrow/pull/11154#issuecomment-919233593

**Comment by bkmgit** [comment]
Thanks. Done. https://issues.apache.org/jira/browse/ARROW-14013

File: docs/source/developers/cpp/building.rst [hunk 1]
```diff
@@ -60,6 +60,17 @@ On Alpine Linux:
            g++ \
            gcc \
            make
+           
+ On Fedora Linux:
```
Comment (kou) [inline docs/source/developers/cpp/building.rst:64]:
```suggestion
On Fedora Linux:
```

File: docs/source/developers/cpp/building.rst [hunk 2]
```diff
@@ -60,6 +60,17 @@ On Alpine Linux:
...
+
+.. code-block:: shell
+
+   sudo dnf install \
+        gcc \
+        gcc-c++ \
+        make \
+        automake \
```
Comment (kou) [inline docs/source/developers/cpp/building.rst:72]: This is needless.

```suggestion
```

File: docs/source/developers/cpp/building.rst [hunk 3]
```diff
@@ -60,6 +60,17 @@ On Alpine Linux:
...
+        cmake
```
Comment (kou) [inline docs/source/developers/cpp/building.rst:73]: Could you sort this list in alphabetical order like others?

File: docs/source/developers/cpp/building.rst (hunk 2 above)
Comment (bkmgit) [reply docs/source/developers/cpp/building.rst:72]: ok

File: docs/source/developers/cpp/building.rst (hunk 3 above)
Comment (bkmgit) [reply docs/source/developers/cpp/building.rst]: Done

**Review by kou — APPROVED** [review]
+1

Thanks!

**Pull request closed without merging**